jwst/extract_2d/nirspec.py

```diff
@@ -221,24 +220,22 @@ def extract_slit(input_model, slit, exp_type):
     """
     slit_wcs = nirspec.nrs_wcs_set_input(input_model, slit.name)
     xlo, xhi, ylo, yhi = offset_wcs(slit_wcs, slit.name)
-    lenshape = len(input_model.data.shape)
-    if lenshape == 2:
-        ext_data = input_model.data[ylo: yhi + 1, xlo: xhi + 1].copy()
-        ext_err = input_model.err[ylo: yhi + 1, xlo: xhi + 1].copy()
-        ext_dq = input_model.dq[ylo: yhi + 1, xlo: xhi + 1].copy()
-        shape = ext_data.shape
-        bounding_box = ((0, shape[1] - 1), (0, shape[0] - 1))
-        ext_var_rnoise = input_model.var_rnoise[ylo: yhi + 1, xlo: xhi + 1].copy()
-        ext_var_poisson = input_model.var_poisson[ylo: yhi + 1, xlo: xhi + 1].copy()
+    ndim = len(input_model.data.shape)
+    if ndim == 2:
+        slit_slice = np.s_[ylo: yhi, xlo: xhi]
+        ext_data = input_model.data[slit_slice].copy()
+        ext_err = input_model.err[slit_slice].copy()
+        ext_dq = input_model.dq[slit_slice].copy()
+        ext_var_rnoise = input_model.var_rnoise[slit_slice].copy()
+        ext_var_poisson = input_model.var_poisson[slit_slice].copy()
         int_times = None
```

Comment:
Since `xlo, xhi, ylo, yhi` are now generated from bounding boxes of the form `((-0.5, 2047.5))`, the result from `_toindex` is `((0, 2048))`.  So no need to add 1 to these anymore.